fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/Utils.java

```diff
@@ -108,13 +107,14 @@ public static String fixSeparatorsToMatchOS(String path) {
   }
   /**
    * This function will produce a fingerprint for the provided schema.
+   *
    * @param schema a schema
    * @return fingerprint for the given schema
    */
-  public static Long getSchemaFingerprint(Schema schema) {
-    Long schemaId = SCHEMA_IDS_CACHE.get(schema);
+  public static int getSchemaFingerprint(Schema schema) {
+    Integer schemaId = SCHEMA_IDS_CACHE.get(schema);
     if (schemaId == null) {
-      schemaId = SchemaNormalization.parsingFingerprint64(schema);
+      schemaId = schema.toString().hashCode();
```

Comment:
Two reasons to avoid this approach:
1. `Schema#toString()` is not stable. Long story, but it even has bugs under old versions of avro (see [AVRO-702](https://issues.apache.org/jira/browse/AVRO-702))
2. `String#hashCode()` isn't necessarily stable either.

We should probably normalize the schema first (you can use  `AvroUtilSchemaNormalization` and write a `AvscWriterPlugin` (see the `FieldLevelPlugin` example - something close to that should work for your use case) to include the junk json you want. [Here's a test](https://github.com/linkedin/avro-util/blob/07050eea0cccc9f665e80a870d6ebc5e130533ed/helper/tests/helper-tests-19/src/test/java/com/linkedin/avroutil1/compatibility/avro19/AvroUtilSchemaNormalizationTest19.java#L342-L361) case you can follow as an example).

However, you can also just use [AvscWriter](https://github.com/linkedin/avro-util/blob/07050eea0cccc9f665e80a870d6ebc5e130533ed/helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvscWriter.java) to serialize the schema if you don't need to normalize.

Reply:
What do you mean by stable? If you mean that it will always yield the same value regardless of Avro version or JVM vendor/version, then that doesn't really matter. We only need this id to be stable within the scope of one runtime of the JVM.

That being said, I can still make it stable in case this ends up being used beyond one runtime in the future.

Reply:
FYI @dg-builder, I have made the ID generation stable in my latest commit.